lib/ex_aws/request/url.ex

```diff
@@ -0,0 +1,23 @@
+defmodule ExAws.Request.Url do
+  @moduledoc false
+
+  @doc """
+  Builds URL for an operation and a config"
+  """
+  def build(operation, config) do
+    query =  operation.params |> URI.encode_query
+    config
+    |> Map.take([:scheme, :host, :port])
+    |> normalize_scheme
+    |> Map.put(:query, query)
+    |> Map.put(:path, operation.path)
+    |> (&struct(URI, &1)).()
+    |> URI.to_string
+  end 
+
+  defp normalize_scheme(%{scheme: scheme} = url) do
+    scheme = scheme |> String.replace("://", "")
+    url |> Map.put(:scheme, scheme)
+  end
+  defp normalize_scheme(url), do: url
+end
```

Comment:
I appreciate moving this into its own module, I think there's a lot of potential here for cleaning up code from other operations. However if we're gonna have a module named `ExAws.Request.URL` we should make sure that it works for all services. If it does not, it belongs under the operation namespace not the request namespace.

Could you try and see if you can use this with all operations?